pkg/k2/language_host/language_host.go

```diff
@@ -19,15 +21,24 @@ import (
 //go:embed python/python_language_host.py
 var pythonLanguageHost string
 
-type ServerAddress struct {
+type ServerState struct {
 	Log     *zap.SugaredLogger
 	Address string
-	HasAddr chan struct{}
+	Error   error
+	Done    chan any
+}
+
+func NewServerState(log *zap.SugaredLogger) *ServerState {
+	return &ServerState{
+		Log:  log,
+		Done: make(chan any),
```

Comment:
nit: `chan any` uses the size of a pointer whereas `chan struct{}` uses zero size. It also uses the types to indicate that it's just a signal, and that no meaningful information other than its completion can be derived from the channel. It's not a big deal, but just explaining why it was that previously.